app/graphql/mutations/move_course.rb

```diff
@@ -0,0 +1,72 @@
+module Mutations
+  class MoveCourse < ApplicationQuery
+    class CourseMustBePresent < GraphQL::Schema::Validator
+      def validate(_object, _context, value)
+        id = value[:id]
+        course = Course.find_by(id: id)
+
+        if course.blank?
+          I18n.t("mutations.update_course.link_not_found_error")
+        end
+      end
+    end
+
+    include QueryAuthorizeSchoolAdmin
+
+    argument :id, ID, required: true
+    argument :direction, Types::MoveDirectionType, required: true
+
+    description "Rearrange courses order"
+
+    validates CourseMustBePresent => {}
+
+    field :success, Boolean, null: false
+
+    def resolve(_params)
+      { success: move_course }
+    end
+
+    def move_course
+      direction = @params[:direction]
+      ordered_courses = Course.includes(:school).order(sort_index: :ASC).to_a
+
+      if direction == "Up"
+        swap_up(ordered_courses, course)
+      else
+        swap_down(ordered_courses, course)
+      end
+
+      ordered_courses.each_with_index do |oc, index|
+        oc.update!(sort_index: index)
+      end
+
+      true
+    end
+
+    private
+
+    def course
+      Course.find_by(id: @params[:id])
+    end
+
+    def resource_school
+      course&.school
+    end
+
+    def swap_up(array, element)
+      index = array.index(element)
+
+      return array if index.nil? || index.zero? || index.blank?
```

Comment:
We are not using the returned array.

```ruby
return if index.blank? || index.zero?
```

Reply:
`swap_up` used in `swap_down`

Reply:
@ramzieus 
When we call `swap_up` inside `move_course` we are supplying `ordered_courses` which is an ActiveRecord array, in this case, the `ordered_courses` array is inside the heap memory which means if we pass it to `swap_up` we are not passing the array but reference to the array. `swap_up` changes the content of `ordered_courses` array with the help of `array` reference parameter. Ultimately, we are not using the return value but making changes directly to the `ordered_courses` array.

Please correct me if I am wrong.

Reply:
@kaisersakhi yes you're right, I just followed the `app/graphql/mutations/move_school_link.rb` code so I didn't focus deeply I thought the team has reviewed it, also there is no spec for the school link mutation.
I'll complete the spec and push the update.

Reply:
> also there is no spec for the school link mutation.

We generally add integration specs and add service, mutation and other specs as per requirement and need.

spec/system/school/courses_index_spec.rb - Integration spec file for `school/courses/` page.